app/abci.go

```diff
@@ -581,6 +581,17 @@ func (app *HeimdallApp) PreBlocker(ctx sdk.Context, req *abci.RequestFinalizeBlo
 		return nil, err
 	}
 
+	milestoneDeletionHeight := helper.GetMilestoneDeletionHeight()
+	if req.Height == milestoneDeletionHeight && milestoneDeletionHeight != 0 {
+		// delete faulty milestone if exists
+		milestoneNumber := helper.GetFaultyMilestoneNumber()
+		if err := app.MilestoneKeeper.DeleteMilestone(ctx, milestoneNumber); err != nil {
```

Comment:
Let's try to be as specific as possible and check for the exact milestone we want to delete. Some conditions I can think of:
1. chain == mainnet
2. milestone id = 1941439
3. "start_block": "76273070",
"end_block": "76273070",
"hash": "eRCiCRhVhnTtuHdZorsIsxrw3g5O7w2JCb51rzWRdI8=",

Reply:
Done via `IsFaultyMilestone` method